paparazzi-gradle-plugin/src/test/projects/configuration-cache-generated-sources/build.gradle

```diff
@@ -0,0 +1,40 @@
+plugins {
+  id 'com.android.library'
+  id 'kotlin-android'
+  id 'app.cash.paparazzi'
+}
+
+android {
+  namespace 'app.cash.paparazzi.plugin.test'
+  compileSdk libs.versions.compileSdk.get() as int
+  defaultConfig {
+    minSdk libs.versions.minSdk.get() as int
+  }
+  compileOptions {
+    sourceCompatibility = libs.versions.javaTarget.get()
+    targetCompatibility = libs.versions.javaTarget.get()
+  }
+  kotlinOptions {
+    jvmTarget = libs.versions.javaTarget.get()
+  }
+}
+
+androidComponents {
+  onVariants(selector().all()) { variant ->
+    def sourceGen =
+      project.tasks.register("do${variant.name.capitalize()}SourceGen", SourceGenTask) {
```

Comment:
Should this have a `doFirst` that actually writes at least an empty source file? Just to prevent accidental NO-SOURCE at any point? e.g. the main source set has no code in this project.

Reply:
what's the concern here?  that subsequent reruns will not be cached due to NO-SOURCE?

Reply:
more like not having a close enough to real life test where a lot of code gets NOT executed due to not having no code. NO-SOURCE is an input check, the outputs will be just empty, so caching-wise it all the same.

Reply:
i see, i try to favor minimal test setups.  the configuration cache bug wasn't surfaced by any actual source file gen, but rather the wiring of tasks/properties in SourceDirectoriesImpl.addGeneratedSourceDirectory, so imo, adding extra config, etc, to make this a real life test adds noise to anyone who reads the test and wants to understand the underlying issue better.